Testscripts/Windows/Multiple-Reboot.ps1

```diff
@@ -0,0 +1,160 @@
+#     the ICA host rather than the VM.
+#
+#     This script will reboot a VM as many times as specified in the count parameter
+#     and check that the VM reboots successfully.
+#
+#     The ICA scripts will always pass the vmName, hvServer, and a
+#     string of testParams to the PowerShell test case script. For
+#     example, if the <testParams> section was written as:
+#
+#
+#     The PowerShell test case scripts need to parse the testParam
+#     string to find any parameters it needs.
+#
+#
+############################################################################
+param([String] $TestParams)
+
+function Main {
+    param (
+        $VMName,
+        $HvServer,
+        $Ipv4,
+        $RootDir,
+        $TestParams
+    )
+    #####################################################################
+    #
+    # Main script body
+    #
+    #####################################################################
+    #
+    # Check input arguments
+    #
+    if ($VMName -eq $null) {
+        LogErr "VM name is null"
+        return "FAIL"
+    }
+
+    if ($HvServer -eq $null) {
+        LogErr "hvServer is null"
+        return "FAIL"
+    }
+    Set-Location $RootDir
+    $vm = Get-VM $VMName -ComputerName $HvServer
+    if (-not $vm) {
+        LogErr "Cannot find VM ${vmName} on server ${hvServer}"
+        return "FAIL"
+    }
+
+    if ($($vm.State) -ne "Running") {
+        LogErr "VM ${vmName} is not in the running state"
+        return "FAIL"
+    }
+
+    LogMsg "Trying to reboot once using ctrl-alt-del from VM's keyboard."
+
+    $VMKB = Get-WmiObject -namespace "root\virtualization\v2" -class "Msvm_Keyboard" -ComputerName $HvServer -Filter "SystemName='$($vm.Id)'"
+    $VMKB.TypeCtrlAltDel()
+
+    if ($? -eq "True") {
+        LogMsg "VM received the ctrl-alt-del signal successfully."
+    }
+    else {
+        LogErr "VM did not receive the ctrl-alt-del signal successfully."
+        return "FAIL"
+    }
+
+    $testCaseTimeout = 120
+
+    while ($testCaseTimeout -gt 0) {
+        if ( (Check-VMState -VMName $VMName -HvServer $HvServer ( "Running" ))) {
+            break
+        }
+        Start-Sleep -seconds 10
+        $testCaseTimeout -= 10
+    }
+
+    while ($testCaseTimeout -gt 0) {
+        if ( (Test-Port -Ipv4addr $Ipv4) ) {
+            break
+        }
+        Start-Sleep -seconds 10
+        $testCaseTimeout -= 10
+    }
+
+    if ($testCaseTimeout -eq 0) {
+        LogErr "Test case timed out waiting for the VM to reach Running state after rebooting with ctrl-alt-del."
+        return "FAIL"
+    }
+
+    #
+    # Set the $bootcount variable and reboot the machine $count times.
+    #
+    LogMsg "Setting the boot count to 0 for rebooting the VM"
+    $bootcount = 0
+
+    while ($count -gt 0) {
+        While ( -not (Test-Port -Ipv4addr $Ipv4) ) {
+            Start-Sleep 5
+        }
+        Restart-VM -VMName $VMName -ComputerName $HvServer -Force
+
+        Start-Sleep 5
+
```

Comment:
check unnecessary new lines in this file too

Reply:
removed